examples/pipeline/wav2letter.py

```diff
@@ -0,0 +1,508 @@
+from torchaudio.models.wav2letter import Wav2Letter
+from torchaudio.transforms import MFCC, Resample
+from tqdm import tqdm
+
+from ctc_decoders import GreedyDecoder, ViterbiDecoder
+from datasets import datasets_librispeech
+from languagemodels import LanguageModel
+from metrics import levenshtein_distance
+
+SIGNAL_RECEIVED = False
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--workers",
+        default=0,
+        type=int,
+        metavar="N",
+        help="number of data loading workers",
+    )
+    parser.add_argument(
+        "--checkpoint",
+        default="",
+        type=str,
+        metavar="PATH",
+        help="path to latest checkpoint",
+    )
+
+    parser.add_argument(
+        "--epochs",
+        default=200,
+        type=int,
+        metavar="N",
+        help="number of total epochs to run",
+    )
+    parser.add_argument(
+        "--start-epoch", default=0, type=int, metavar="N", help="manual epoch number"
+    )
+    parser.add_argument(
+        "--print-freq",
+        default=10,
+        type=int,
+        metavar="N",
+        help="print frequency in epochs",
+    )
+    parser.add_argument(
+        "--progress-bar", action="store_true", help="use progress bar while training"
+    )
+
+    parser.add_argument(
+        "--decoder",
+        metavar="D",
+        default="greedy",
+        choices=["greedy", "viterbi"],
+        help="decoder to use",
+    )
+
+    parser.add_argument(
+        "--batch-size", default=64, type=int, metavar="N", help="mini-batch size"
+    )
+
+    parser.add_argument(
+        "--n-bins",
+        default=13,
+        type=int,
+        metavar="N",
+        help="number of bins in transforms",
+    )
+    parser.add_argument(
+        "--learning-rate",
+        default=1.0,
+        type=float,
+        metavar="LR",
+        help="initial learning rate",
+    )
+    parser.add_argument(
+        "--gamma",
+        default=0.96,
+        type=float,
+        metavar="GAMMA",
+        help="learning rate exponential decay constant",
+    )
+    # parser.add_argument('--momentum', default=0.9, type=float, metavar='M', help='momentum')
+    parser.add_argument(
+        "--weight-decay", default=1e-5, type=float, metavar="W", help="weight decay"
+    )
+    parser.add_argument("--eps", metavar="EPS", type=float, default=1e-8)
+    parser.add_argument("--rho", metavar="RHO", type=float, default=0.95)
+    parser.add_argument("--clip-norm", metavar="NORM", type=float, default=0.0)
+
+    parser.add_argument(
+        "--dataset",
+        default="librispeech",
+        type=str,
+        help="select dataset to train with",
+    )
+    parser.add_argument(
+        "--distributed", action="store_true", help="enable DistributedDataParallel"
+    )
+    parser.add_argument("--jit", action="store_true", help="if used, model is jitted")
+
+    args = parser.parse_args()
+
+    return args
+
+
+def signal_handler(a, b):
```

Comment:
The need for functions like this worry me, because I'd imagine most users to not be aware of their necessity or purpose.